pkg/controller/servicebindingrequest/reconciler.go

```diff
@@ -329,6 +355,70 @@ func (r *Reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 	// splitting instance from it's status
 	sbrStatus := &sbr.Status
 
+	if sbrStatus.Conditions == nil {
+		logger.Infor("No Condition on Status has been reported")
+		conditionsv1.SetStatusCondition(&sbrStatus.conditions, conditionsv1.Condition{
+			Type: conditions.BindingReady
+			Status: conditions.ConditionFlase
+			Reason: "Binding Conditions"
+			Message: "Binding has no conditions"
+		})
+		conditionsv1.SetStatusCondition(&sbrStatus.conditions, conditionsv1.Condition{
+			Type: conditions.BindingInProgress
+			Status: conditions.ConditionTrue
+			Reason: "Binding Conditions"
+			Message: "Binding has no conditions"
+		})
+		conditionsv1.SetStatusCondition(&sbrStatus.conditions, conditionsv1.Condition{
+			Type: conditions.BindingFailed
+			Status: conditions.ConditionFlase
+			Reason: "Binding Conditions"
+			Message: "Binding has no conditions"
+		})
+	} else {
+		for _, condition := range sbrStatus.Conditions{
```

Comment:
What is the intention of this loop?Extract this to a function.

Reply:
I thought that when reconcile begins we can keep standard conditions with some standard (predefined values). Shall we omit this if and else block completely?

Reply:
Existing conditions should be loaded since existing information is used to compute whether a transition has happened.

Take, for example, the following pre-existing condition data:

```yaml
conditions:
- type: Ready
  status:` False
  reason: SomeReason
  message: "Details go here for humans to consume"
  lastTransitionTimestamp: ...
```
Assuming reconcile will update this condition, but with another reason this time:

```yaml
conditions:
- type: Ready
  status: False
  reason: SomeOtherReason
  message: "Other set of details"
```

In this case, `lastTransitionTimestamp` will not update the last transition timestamp, since no transition has been recorded. Now, in the case `status` transitions from `False` to `True`, the transition timestamp should be recorded:

```yaml
conditions:
- type: Ready
  status: True
  lastTransitionTimestamp: ...
```
It is important to mention that a missing condition represents the status `Unknown`, meaning that whenever a condition is recorded a transition from `Unknown` to either `True` or `False` has happened.